### PR TITLE
fix: python build failed in openeuler

### PR DESCRIPTION
**Pull request by pkking**

### CODE REVIEW

File: python/python-copr.spec [hunk 1]
```diff
@@ -147,6 +147,7 @@ BuildRequires: python3-pytest
 BuildRequires: python3-requests-gssapi
 BuildRequires: python3-future
 BuildRequires: python3-filelock
+BuildRequires: pyproject-rpm-macros
```
Comment (praiskup) [inline python/python-copr.spec:150]: This seems redundant.  Per guidelines, we need python3-devel:
https://docs.fedoraproject.org/en-US/packaging-guidelines/Python/

File: python/python-copr.spec [hunk 2]
```diff
@@ -137,7 +137,7 @@ Requires: python3-future
 
 %{?python_provide:%python_provide python3-copr}
 
-%if 0%{?fedora} > 30 || 0%{?rhel} > 8
+%if 0%{?fedora} > 30 || 0%{?rhel} > 8 || 0%{?openEuler}
```
Comment (praiskup) [inline python/python-copr.spec:140]: This seems like an ideal candidate for '%else' branch above to `%if 0%{?rhel} && 0%{?rhel} <= 8`

File: python/python-copr.spec (hunk 1 above)
Comment (praiskup) [reply python/python-copr.spec:150]: Hmm, I am not sure why we actually need all the other requirements that should be automatically generated.

File: python/python-copr.spec (hunk 2 above)
Comment (pkking) [reply python/python-copr.spec]: > This seems like an ideal candidate for '%else' branch above to `%if 0%{?rhel} && 0%{?rhel} <= 8`

Done/

File: python/python-copr.spec (hunk 1 above)
Comment (praiskup) [reply python/python-copr.spec:149]: ACK.  I think we can keep it (seems like a distro glitch).

File: python/python-copr.spec [hunk 3]
```diff
@@ -1,6 +1,6 @@
 %global srcname copr
 
-%if 0%{?fedora} || 0%{?rhel} > 7
+%if 0%{?fedora} || 0%{?rhel} > 7 || 0%{?openEuler}
```
[the remark this answers is not in the document]
Comment (pkking) [reply python/python-copr.spec]: > Would you mind using `!0%{?rhel} || 0%{?rhel} > 7` ? Any new distro will directly use python3.

Done

**Review by praiskup — APPROVED** [review]

**Pull request merged**